pkg/operator/controller/controller.go

```diff
@@ -326,6 +316,76 @@ func (r *reconciler) ensureInternalRouterServiceForIngress(deployment *appsv1.De
 	return nil
 }
 
+// ensureRouterNamespaceAsset ensures that the router namespace exists and
+// matches the expected namespace.
+func (r *reconciler) ensureRouterNamespaceAsset() error {
+	ns, err := r.ManifestFactory.RouterNamespace()
+	if err != nil {
+		return fmt.Errorf("failed to build router namespace: %v", err)
+	}
+
+	namespaceMatcher := func(obj1, obj2 runtime.Object) bool {
+		ns1 := obj1.(*corev1.Namespace)
+		ns2 := obj2.(*corev1.Namespace)
+
+		// TODO: Should we instead just check that the
+		//       "openshift.io/run-level" label matches?
+		if !reflect.DeepEqual(ns1.Labels, ns2.Labels) {
+			return false
+		}
+
+		// TODO: Should we instead just check that the
+		//       "openshift.io/node-selector" annotation matches?
+		if !reflect.DeepEqual(ns1.Annotations, ns2.Annotations) {
```

Comment:
I don't think we need to check individual labels/annotations. This will restrict users to add custom labels/annotations on the namespace and I think that is good, it avoids side effects caused by some annotations (for e.g. node selector on namespace will filter the nodes before node selector on pods is applied).
I feel any custom label/annotation should be part of custom resource to ensure consistent behavior.

Reply:
Ok, I will remove those TODOs.